### PR TITLE
Move row_data_tracked to ModelExt trait

### PR DESCRIPTION
**Pull request by lukas-jung**
The tests pass but I'm not sure if it is accessible from the api.

### CODE REVIEW

File: internal/core/model.rs [hunk 1]
```diff
@@ -1204,7 +1208,7 @@ fn test_vecmodel_set_vec() {
     }
 
     let view = Rc::pin(TestView::default());
-
+    //iaosjdfk
```
Comment (ogoffart) [inline internal/core/model.rs:1211]: !!!

**Review by ogoffart — COMMENTED** [review]
We need to re-export it from api/rs/slint/lib.rs

We can also even add a `pub use slint::ModelExt as _` to the generated code in rust.rs  (line 240)

File: internal/core/model.rs (hunk 1 above)
Comment (lukas-jung) [reply internal/core/model.rs:1211]: ahh. I just fished this out of the stash. should have checked it better.

**Comment by ogoffart** [comment]
The context is that we're thinking about adding more method to the Model trait such as `filter` and `map`, but these functions don't need to be re-implemented by the users, the default implementation is the correct one. This is also the case for row_data_tracked that was added in this release cycle.  The idea is to make an extension trait with all these method instead of putting them in the Model trait.  Since row_data_tracked was added in this release, we can move it now

**Review by hunger — APPROVED** [review]
Makes sense to me

**Pull request merged**